app/controllers/decks_controller.rb

```diff
@@ -0,0 +1,58 @@
+class DecksController < ApplicationController
+  before_action :set_deck, only: [ :show, :edit, :update, :destroy ]
+  helper_method :current?
+
+  def index
+    @decks = current_user.decks.all
+  end
+
+  def show
+    @cards = @deck.cards.all
+  end
+
+  def new
+    @deck = current_user.decks.build
+  end
+
+  def edit
+  end
+
+  def create
+    @deck = current_user.decks.build(deck_params)
+
+    if @deck.save
+      redirect_to @deck
+      flash[:notice] = 'Новая колода добавлена'
+    else
+      render 'new'
+    end
+  end
+
+  def update
+    if @deck.update(deck_params)
+      redirect_to @deck
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    @deck.destroy
+
+    redirect_to decks_path
+  end
+
+  def current?(deck_id)
+    current_user.current_deck_id == deck_id

```

Comment:
why you need this in controller? Why not in User model write instance method, like
```
def current_deck?(deck)
  current_user.current_deck == deck
end
```